vms/platformvm/txs/fee/complexity.go

```diff
@@ -149,29 +148,26 @@ var (
 			wrappers.IntLen + // validator rewards typeID
 			wrappers.IntLen + // delegator rewards typeID
 			wrappers.IntLen, // delegation shares
-		gas.DBRead:  1,
-		gas.DBWrite: 1,
-		gas.Compute: 0,
+		gas.DBRead:  1, // get staking config
+		gas.DBWrite: 3, // put current staker + write weight diff + write pk diff
 	}
 	IntrinsicAddPermissionlessDelegatorTxComplexities = gas.Dimensions{
 		gas.Bandwidth: IntrinsicBaseTxComplexities[gas.Bandwidth] +
 			intrinsicValidatorBandwidth + // validator
 			ids.IDLen + // subnetID
 			wrappers.IntLen + // num stake outs
 			wrappers.IntLen, // delegator rewards typeID
-		gas.DBRead:  1,
-		gas.DBWrite: 1,
-		gas.Compute: 0,
+		gas.DBRead:  1, // get staking config
+		gas.DBWrite: 2, // put current staker + write weight diff
```

Comment:
don't both this and add validator happen via `putStaker` ?  why not pk diff here too?

Reply:
Delegator changes can never result in the weight going from `x -> 0` or `0 -> x` so it can never modify the public key.